src/saveToCase.js

```diff
@@ -632,7 +632,7 @@ define([
                 if (caseIdRegex.test(value.ref)) {
                     mug.p.case_id = value.value;
                     mug.form.dropSetValues(function(inner) {
-                        return value.value === inner.value;
+                        return value.ref === inner.ref;
```

Comment:
Should this also be checking `value.event === inner.event` as well? `ref` and `event` are the two things I think of as making a `setvalue` node "unique" (although they need not actually be unique because ordering of `setvalue`s in the form is significant as well).